crates/compiler/parse/src/expr.rs

```diff
@@ -666,185 +679,261 @@ pub fn parse_single_def<'a>(
-                        increment_min_indent(expr_start(options)),
-                        EExpr::IndentEnd,
-                    );
-
-                    let (_, loc_def_expr, state) =
-                        parse_def_expr.parse(arena, state, min_indent)?;
-                    let value_def =
-                        ValueDef::Body(arena.alloc(loc_pattern), &*arena.alloc(loc_def_expr));
-                    let region = Region::span_across(&loc_pattern.region, &loc_def_expr.region);
-
-                    Ok((
-                        MadeProgress,
-                        Some(SingleDef {
-                            type_or_value: Either::Second(value_def),
-                            region,
-                            spaces_before: spaces_before_current,
-                        }),
-                        state,
-                    ))
-                }
-                Ok((_, BinOp::IsAliasType, state)) => {
-                    // the increment_min_indent here is probably _wrong_, since alias_signature_with_space_before does
-                    // that internally.
-                    // TODO: re-evaluate this
-                    let parser = increment_min_indent(alias_signature_with_space_before());
-                    let (_, ann_type, state) = parser.parse(arena, state, min_indent)?;
-                    let region = Region::span_across(&loc_pattern.region, &ann_type.region);
-
-                    match &loc_pattern.value.extract_spaces().item {
-                        Pattern::Apply(
-                            Loc {
-                                value: Pattern::Tag(name),
-                                ..
-                            },
-                            alias_arguments,
-                        ) => {
-                            let name = Loc::at(loc_pattern.region, *name);
-                            let header = TypeHeader {
-                                name,
-                                vars: alias_arguments,
-                            };
+            // This may be a def or alias.
+            let operator_result = operator().parse(arena, state.clone(), min_indent);
 
-                            let type_def = TypeDef::Alias {
-                                header,
-                                ann: ann_type,
-                            };
+            if let Ok((_, BinOp::Assignment, operator_result_state)) = operator_result {
+                let result = parse_single_def_assignment(
+                    options,
+                    // to support statements we have to increase the indent here so that we can parse a child def
+                    // within a def and still continue to parse the final expression for this def
+                    // e.g.
+                    // main =
+                    //     Stdout.line! "Bar"
+                    //     a=Stdout.line! "Foo"
+                    //     Task.ok {}
+                    operator_result_state.line_indent() + 1,
```

Comment:
This was using `min_indent`, however it took me some time to figure out that I can use `line_indent`. This seems to give the desired behaviour. It reduces the ability to parse poorly indented code, specifically if there is not enough spaces before the expression.